frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.unit.spec.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState } from "react";
+import userEvent from "@testing-library/user-event";
+import { waitFor, within, renderWithProviders, screen } from "__support__/ui";
+import { setupSearchEndpoints } from "__support__/server-mocks";
+import { createMockSearchResult } from "metabase-types/api/mocks";
+import { SearchFilterModal } from "metabase/search/components/SearchFilterModal/SearchFilterModal";
+import { SearchModelType } from "metabase-types/api";
+import { SearchFilters } from "metabase/search/types";
+
+const TestSearchFilterModal = ({
+  initialFilters = {},
+  onChangeFilters,
+}: {
+  initialFilters?: SearchFilters;
+  onChangeFilters: jest.Mock;
+}) => {
+  const [filters, setFilters] = useState(initialFilters);
+
+  onChangeFilters.mockImplementation(newFilters => setFilters(newFilters));
+
+  return (
+    <SearchFilterModal
+      isOpen={true}
+      setIsOpen={jest.fn()}
+      value={filters}
+      onChangeFilters={onChangeFilters}
+    />
+  );
+};
+
+const TEST_TYPES: Array<SearchModelType> = [
+  "action",
+  "card",
+  "collection",
+  "dashboard",
+  "database",
+  "dataset",
+  "metric",
+  "pulse",
+  "segment",
+  "table",
+];
+
+const TEST_INITIAL_FILTERS: SearchFilters = {
+  type: TEST_TYPES,
```

Comment:
One thing I'm confusing seeing this the first time is why `type` represents a list of type. Shouldn't this be `types`? Could you help explain this a bit?